replica/compaction_group.hh

```diff
@@ -233,6 +234,7 @@ protected:
     // The list entries are unlinked automatically when the storage group, they belong to, is removed.
     compaction_group_list _compaction_groups;
     storage_group_map _storage_groups;
+    seastar::rwlock _lock;
```

Comment:
Please add rules for when the lock should be held (using what type of lock), and when the lock is unnecessary. It should be clear what the lock protects.